### PR TITLE
Created the teacher & student and person classes

### PR DESCRIPTION
**Pull request by OpondoG**
In this project, I will start building my school library app. In this initial step, I will implement the classes to represent students and teachers.

- Create class Person with the following:

Instance vars: @id, @name, and @Age.
Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
Getters for @id, @name, and @Age.
Setters for @name and @Age.
Private method is_of_age? that returns true if @Age is greater or equal to 18 and false otherwise.
Public method can_use_services? that returns true if person is of age or if they have permission from parents.

- Create class Student with the following:

Inherits from Person.
Constructor extends parent's constructor by adding @classroom and a parameter for it.
Method play_hooky that returns "¯(ツ)/¯".

- Create class Teacher with the following:

Inherits from Person.
Constructor extends parent's constructor by adding https://github.com/specialization and a parameter for it.
Override can_use_services? so it always returns true.
Each class should be saved in a separate file.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-    @classroom = classroom
+  def initialize(age, class_room, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @class_room = class_room
   end
 
   def play_hooky
-    '¯\\(ツ)/¯'
+    '¯\(ツ)/¯'
   end
```
Comment (EricMbouwe) [inline student.rb]: - [x] With the association between `Classroom` and `Student` you would have a method where a student is added to the classrooms' students. Let's add it

File: person.rb [hunk 2]
```diff
@@ -1,27 +1,16 @@
...
-
-  def initialize(name, age, parent_permission: true)
-    super()
+class Person
+  def initialize(age, name = 'Unknown', parent_permission = 'true')
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
   end
```
Comment (EricMbouwe) [inline person.rb]: - [x] A method for adding rentals should be present. You can refer to this [lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) to learn more

**Review by EricMbouwe — CHANGES_REQUESTED** [review]
Hi @OpondoG, 

Good job so far 🔥!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 🔥🎉
- PR ✔
- No linter errors ✔

## Required Changes ♻️
- [x] There is no class called `Classroom`. Let's add it
- [x] Make sure the `has-many` / `belongs-to` relationships between `Classroom` and `Student` is created
- [x] There is no class called `Book`. Let's add it
- [x] There is no class called `Rental`. Let's add it
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: student.rb (hunk 1 above)
Comment (danmainah) [reply student.rb]: - [ ] @OpondoG as previously requested kindly add a method to add a student to the classroom to achieve the belongs-to association.

File: person.rb (hunk 2 above)
Comment (danmainah) [reply person.rb]: - [x] @OpondoG as previously requested kindly add a method to add rentals in your person class.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @OpondoG , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used


 # Required Changes ♻️

- [x] Kindly add the classroom class as previously requested also make sure the has-many / belongs-to relationships between Classroom and Student is created, this is a project requirement
image

- [x] Kindly add the rental and book class as previously requested, this is a project requirement
image



_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
```
Comment (Ghiftee) [inline classroom.rb:2]: - [x] To complete the relationship between classroom and student, kindly create getter method for student. e.g

``` attr_reader :students ```

File: student.rb [hunk 4]
```diff
@@ -1,14 +1,12 @@
...
 class Student < Person
-  attr_accessor :classroom
-
-  def initialize(name, age, classroom, parent_permission: true)
-    super(name, age, parent_permission: parent_permission)
-    @classroom = classroom
+  def initialize(age, class_room, name = 'Unknown', parent_permission: true)
+    super(age, name, parent_permission)
+    @class_room = class_room
   end
```
Comment (Ghiftee) [inline student.rb:10]: -  [x] Like the previous reviewers requested, kindly add a method which satisfies the belongs-to side of the Classroom-student relationship. e.g

``````ruby
attr_reader :classroom
 
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end

File: person.rb [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
+  def initialize(age, name = 'Unknown', parent_permission = 'true')
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
 
+  attr_accessor :name, :age
```
Comment (Ghiftee) [inline person.rb]: - [x] Before rentals can be accessed, it needs to be both readable and editable, kindly add ```rentals``` to ```attr_accessor```
- [x] Also, I suggest you move this code to the top of the file as it is best practice.

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there 👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
✔️ The classroom class has been added
✔️ A method for adding rentals is present
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
_Check the comments under the review._

- [ ] Like previous reviewers requested, kindly create a class for ```book``` and ```rental```

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @OpondoG,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- No linter errors ✔️
- Well organized and clean project ✔️
approval

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by OpondoG** [comment]
Hi @uwadonat  ,🥳

Thank you for taking the time to review my code. I really appreciate your feedback and insights.
Thank you again for your help.

**Pull request merged**